### PR TITLE
make conversion to identical type a nop

### PR DESCRIPTION
**Pull request by KristofferC**
fixes https://github.com/eschnett/SIMD.jl/issues/5#issuecomment-207091816

### CODE REVIEW

File: src/SIMD.jl [hunk 1]
```diff
@@ -121,7 +121,8 @@ end
 (::Type{Vec}){N,T<:ScalarTypes}(xs::NTuple{N,T}) = Vec{N,T}(xs)
 
 # Convert between vectors
-@inline Base.convert{N,T}(::Type{Vec{N,T}}, v::Vec{N}) = Vec{N,T}(Tuple(v))
+@inline Base.convert{N,T1, T2}(::Type{Vec{N,T1}}, v::Vec{N, T2}) = Vec{N,T1}(Tuple(v))
```
Comment (eschnett) [inline src/SIMD.jl]: I don't think it's necessary to introduce the type `T2` here; the original function could remain as is.

Did you look at the generated code -- is it actually smaller / faster?

**Comment by eschnett** [comment]
Can you add tests as well? I try to have at least one test for every function.

File: src/SIMD.jl (hunk 1 above)
Comment (KristofferC) [inline src/SIMD.jl]: I first did it like that but then the second function didn't get called at all. I would have thought the second function would be more specific but apparently not. I have had similar things like this happened before and just specifying all parameters usually solves it.

**Comment by KristofferC** [comment]
Like this?

**Pull request closed without merging**